core/common/src/main/java/alluxio/ParameterizedPropertyKey.java

```diff
@@ -0,0 +1,126 @@
+   * different parameters. E.g., * {@code Template.MASTER_TIERED_STORE_GLOBAL_LEVEL_ALIAS.format(0)}
+   */
+  @ThreadSafe
+  public enum Template {
+    MASTER_MOUNT_TABLE_ENTRY_ALLUXIO("alluxio.master.mount.table.%s.alluxio",
+        "alluxio\\.master\\.mount\\.table.(\\w+)\\.alluxio"),
+    MASTER_MOUNT_TABLE_ENTRY_OPTION("alluxio.master.mount.table.%s.option",
+        "alluxio\\.master\\.mount\\.table\\.(\\w+)\\.option"),
+    MASTER_MOUNT_TABLE_ENTRY_OPTION_PROPERTY("alluxio.master.mount.table.%s.option.%s",
+        "alluxio\\.master\\.mount\\.table\\.(\\w+)\\.option(\\.\\w+)++"),
+    MASTER_MOUNT_TABLE_ENTRY_READONLY("alluxio.master.mount.table.%s.readonly",
+        "alluxio\\.master\\.mount\\.table\\.(\\w+)\\.readonly"),
+    MASTER_MOUNT_TABLE_ENTRY_SHARED("alluxio.master.mount.table.%s.shared",
+        "alluxio\\.master\\.mount\\.table\\.(\\w+)\\.shared"),
+    MASTER_MOUNT_TABLE_ENTRY_UFS("alluxio.master.mount.table.%s.ufs",
+        "alluxio\\.master\\.mount\\.table\\.(\\w+)\\.ufs"),
+    MASTER_TIERED_STORE_GLOBAL_LEVEL_ALIAS("alluxio.master.tieredstore.global.level%d.alias",
+        "alluxio\\.master\\.tieredstore\\.global\\.level(\\d+)\\.alias"),
+    WORKER_TIERED_STORE_LEVEL_ALIAS("alluxio.worker.tieredstore.level%d.alias",
+        "alluxio\\.worker\\.tieredstore\\.level(\\d+)\\.alias"),
+    WORKER_TIERED_STORE_LEVEL_DIRS_PATH("alluxio.worker.tieredstore.level%d.dirs.path",
+        "alluxio\\.worker\\.tieredstore\\.level(\\d+)\\.dirs\\.path"),
+    WORKER_TIERED_STORE_LEVEL_DIRS_QUOTA("alluxio.worker.tieredstore.level%d.dirs.quota",
+        "alluxio\\.worker\\.tieredstore\\.level(\\d+)\\.dirs\\.quota"),
+    WORKER_TIERED_STORE_LEVEL_RESERVED_RATIO("alluxio.worker.tieredstore.level%d.reserved.ratio",
+        "alluxio\\.worker\\.tieredstore\\.level(\\d+)\\.reserved\\.ratio"),
+    ;
+
+    private final String mFormat;
+    private final Pattern mPattern;
+
+    /**
+     * Constructs a property key format.
+     *
+     * @param format String of this property as formatted string
+     * @param re String of this property as regexp
+     */
+    Template(String format, String re) {
+      mFormat = format;
+      mPattern = Pattern.compile(re);
+    }
+
+    @Override
+    public String toString() {
+      return Objects.toStringHelper(this).add("format", mFormat).add("pattern", mPattern)
+          .toString();
+    }
+
+    /**
+     * Converts a property key template (e.g.,
+     * {@link #WORKER_TIERED_STORE_LEVEL_RESERVED_RATIO}) to a {@link PropertyKey} instance.
+     *
+     * @param params ordinal
+     * @return corresponding property
+     */
+    public ParameterizedPropertyKey format(Object... params) {
+      return new ParameterizedPropertyKey(String.format(mFormat, params));
+    }
+  }
+
+  /**
+   * @param input string of property key
+   * @return whether the input is a valid property name
+   */
+  public static boolean isValid(String input) {
```

Comment:
same. It must be `public` as its base class `PropertyKey` has a public static method with same name. Dropping access privileges is disallowed here.